openmdao/devtools/iprofile_app/iprofile_app.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import print_function
+
+import os
+import pstats
+import sys
+import traceback
+import time
+import webbrowser
+import fnmatch
+import threading
+import argparse
+import json
+from six import StringIO, iteritems, itervalues
+import tornado.ioloop
+import tornado.web
+from collections import defaultdict, deque
+from itertools import groupby
+
+from openmdao.devtools.iprofile import process_profile, profile_py_file
+from openmdao.devtools.iprof_utils import func_group, find_qualified_name, _collect_methods
+
+
+def launch_browser(port):
+    time.sleep(1)
+    webbrowser.get().open('http://localhost:%s' % port)
+
+def startThread(fn):
+    thread = threading.Thread(target=fn)
+    thread.setDaemon(True)
+    thread.start()
+    return thread
+
+def _parent_key(d):
+    parts = d['id'].rsplit('-', 1)
+    if len(parts) == 1:
+        return ''
+    return parts[0]
+
+def stratify(call_data, sortby='time'):
+    """
```

Comment:
Maybe being pedantic here, but since this tool is in our feature reference, it is user-facing, so we really should follow all of our PEP/etc coding standards for docstrings and everything else.

Reply:
The only user facing functions for any of the tracers/profilers are setup(), start() and stop(), and those have the proper docstrings.  I did go through and make a bunch of stuff private and added some docstrings here and there.